client/rpc.go

```diff
@@ -152,23 +165,33 @@ func (r *RPC) handleConnections() {
 			continue
 		}
 		r.Log().Trace("client received: ", string(data))
-		var msg wire.Result
-		if err := json.Unmarshal(data, &msg); err != nil {
-			r.Log().Error("decoding message: ", err)
-			continue
-		}
 
-		switch {
-		case msg.ID != "":
-			r.callCallback(msg, data)
-		case msg.Topic != "":
-			if r.subscription == nil {
-				r.Log().Error("Received proof without subscription")
-				return
+		if !hasConfig {
+			if err := json.Unmarshal(data, &r.clientCfg); err != nil {
+				r.Log().Error("decoding client config: ", err)
+				continue
+			}
+			close(r.hasConfig)
+			hasConfig = true
+		} else {
```

Comment:
I think it's easier to read if you add a `continue` in the first `if` branch and then put the normal path of execution after the `if`, instead of having it inside the `else` branch, which is always executed except for the very first msg.